Source/Core/Core/HW/DVDInterface.cpp

```diff
@@ -1291,7 +1343,7 @@ u64 SimulateDiscReadTime(u64 offset, u32 length)
 		}
 	}
 
-	g_last_read_offset = (offset + length - 2048) & ~2047;
+	g_last_read_offset = end_offset;
```

Comment:
I'm pretty sure that doesn't match what the hardware does., but feel free to prove me wrong.

If you want to experiment, my old DVD testing code I used to discover the buffering is here: https://github.com/magumagu/hwtests/blob/dvdhack/gxtest/source/main.cpp#L3001 .  Should be straightforward to compile/run/modify.  You can ping me on IRC if you have questions, or if you need me to run it for some reason.

There's one important question here: does the DVD drive actually perform decryption, or does IOS do it?  If IOS does it, there's probably another layer of caching we need to account for to compute correct timings.